torch_geometric/data/remote_backend.py

```diff
@@ -0,0 +1,72 @@
+# This file defines a set of utilities for remote backends (backends that are
+# characterize as Tuple[FeatureStore, GraphStore]).
+from typing import Tuple, Union
+
+from torch_geometric.data.feature_store import FeatureStore
+from torch_geometric.data.graph_store import GraphStore
+from torch_geometric.typing import EdgeType, NodeType
+
+
+def num_nodes(
+    feature_store: FeatureStore,
+    graph_store: GraphStore,
+    query: Union[NodeType, EdgeType],
+) -> Union[int, Tuple[int, int]]:
+    r"""Returns the number of nodes in the node type or the number of source
+    and destination nodes in an edge type by sequentially accessing attributes
+    in the feature and graph stores that reveal this number. Returns a single
+    integer in the case of a feature store and a tuple of integers in the case
+    of a graph store."""
+    def _matches_edge_type(query: Union[NodeType, EdgeType],
+                           edge_type: EdgeType) -> bool:
+        if isinstance(query, (list, tuple)):  # EdgeType
+            return query == edge_type
+        else:
+            return query == edge_type[0] or query == edge_type[-1]
+
+    def _matches_node_type(query: Union[NodeType, EdgeType],
+                           node_type: NodeType) -> bool:
+        if isinstance(query, (list, tuple)):  # EdgeType
+            return query[0] == node_type or query[-1] == node_type
+        else:
+            return query == node_type
+
+    node_query = isinstance(query, NodeType)
+
+    # 1. Check GraphStore:
+    edge_attrs = graph_store.get_all_edge_attrs()
+    for edge_attr in edge_attrs:
+        if (_matches_edge_type(query, edge_attr.edge_type)
+                and edge_attr.size is not None):
+            if node_query:
+                return edge_attr.size[0] if query == edge_attr.edge_type[
+                    0] else edge_attr.size[1]
+            else:
+                return edge_attr.size
+
+    # 2. Check FeatureStore:
```

Comment:
Check `FeatureStore` first and then `GraphStore`. When the last node is an isolated node the `FeatureStore` will have a more accurate size.

Reply:
I think we want to check GraphStore first, since the sizes of the edge indices are guaranteed to be accurate. We can then back off to features if we have to. If the last node is an isolated node, we just won't find it in the GraphStore, so we will back off to the FeatureStore anyway, right?